.github/workflows/code-quality.yml

```
@@ -0,0 +1,68 @@
+
+      - name: Run unused dependency check
+        run: mix deps.unlock --check-unused
+        continue-on-error: true
+
+      - name: Check for security vulnerabilities
+        run: mix sobelow --config --format json --out sobelow-report.json || true
+        continue-on-error: true
+
+      - name: Generate code quality report
+        if: always()
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2129:style:1:1: Consider using { cmd1; cmd2; } >> file instead of individual redirects

(shellcheck)